### ISSUE TITLE
Parsing zero mass bodies

### ISSUE THREAD

**Issue by amcastro-tri** [RobotLocomotion/drake#13903]
This is from a [Slack discussion](https://drakedevelopers.slack.com/archives/C2WBPQDB7/p1597783243328300).

When using SDF parsing @avalenzu gets the "The resulting spatial inertia ... is not physically valid" error. However he can create a zero spatial inertia with mass=0, p_PScm_E=[0, 0, 0] and G_SP_E=UnitInertia([0, 0, 0]).

### The problem

`ExtractSpatialInertiaAboutBoExpressedInB()` in `detail_sdf_parser.cc` makes use of `MakeFromCentralInertia()`, which assumes `mass > 0`

**Note I**, zero spatial inertias are supported in our math, that is why programmatically creating zero spatial inertias is allowed.

**Note II**, `MakeFromCentralInertia()` assumes `mass>0`, however this is missing proper docs and throws from somwehere else buried in the code.

### A solution

- [ ] We can make `ExtractSpatialInertiaAboutBoExpressedInB()` to programmatically create a zero spatial inertia as Andres did when we have `mass = 0`. (@avalenzu)

- [ ] `MakeFromCentralInertia()` should throw and document the assumption `mass > 0`. (@mitiguy)

### Question

- Do we have the same problem in URDF?
  - Answer: ~Yes, the URDF makes an analogous call to `MakeFromCentralInertia()`. ~ _edit: Actually, URDF has an even more streamlined convention for this. The parser branches to the zero spatial inertia case if the `<inertial>` block is omitted._
- Does sdformat allow `m=0`? (I (Alejandro) remember at some point in the past there was some hardcoded minimum value).

**Comment by avalenzu** [comment]
@mitiguy, as per the discussion @amcastro-tri mentioned, I've assigned you for the doc/throw fix to `MakeFromCentralInertia()`.

**Comment by avalenzu** [comment]
@amcastro-tri, in the case of `mass == 0` and non-zero moments/products of inertia, should we take the "zero spatial inertia" branch? Or call `MakeFromCentralInertia()` and let it fail? I ask because the SDFormat default for `<inertia/>` is `ixx = iyy = izz = iyz = ixz = ixy = 1`, so a tag like this:
```.xml
<inertial>
  <mass>0</mass>
</inertial>
```
would give us zero mass and non-zero moments/products. Given that the default for `<mass>` is 1 (i.e. to get `mass == 0` the file has to specify it explicitly), I'd prefer to take the "zero spatial inertia" branch.

**Comment by amcastro-tri** [comment]
I actually concur with you, if I say `<mass>0</mass>` only, I'd personally like to take the zero spatial inertia branch.

On further thought, it'd seem wrong, since non-zero inertia components Ixx, etc. would imply non-zero mass. In the end, I believe we should abide by the SDF specification, which now says nothing about this.

Maybe a middle ground would be to make it now so that we do need to say `Ixx=Iyy=etc=0` and work with OSRC to include this case in the spec?

**Comment by avalenzu** [comment]
OK, I'll make it so that we take the zero spatial inertia branch iff the mass and all components of inertia are zero.

**Comment by sherm1** [comment]
Can we make the case mass=0, inertia≠0 an error? It doesn't make any physical sense.

**Comment by amcastro-tri** [comment]
I agree with you @sherm1, but it'd definitely be nice if `<mass>0</mass>` alone meant "zero mass body". But that I believe should be part of the actual SDF spec.

**Comment by jwnimmer-tri** [comment]
I'm having a difficult time following this discussion.  Did #13908 resolve the problem?  If not, what is the unresolved victory condition?

**Comment by sherm1** [comment]
From above it looks like `SpatialInertia::MakeFromCentralInertia()` still needs to document the requirement that mass > 0 and might need a better error message in that case? What do you think @mitiguy (read above discussion)?

**Comment by jwnimmer-tri** [comment]
@mitiguy please share your thoughts.

**Comment by mitiguy** [comment]
I agree with Rico.  I have be looking at the URDF/SDF parser code and associated tests and think that the testing for the mass/inertia conditions centralized, so each format can leverage the general code.  

I think the tests should be as exhaustive as possible and the error messages as instructive as possible. 

 In addition, I am working with Open Robotics to update the <inertial> tag and sub-tags as the specification is ambiguous.  Lastly, I have investigated the URDF/SDF from the CAD packages OnShape and am doing the same from SolidWorks.  I have found inconsistencies with our Drake convention and think we need to clearly document our product of inertia conventions.
